src/App.jsx

```diff
@@ -14,33 +16,90 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+const SORT_FIELD_BY_ALPHABET = 'alphabet';
+const SORT_FIELD_BY_LENGTH = 'length';
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+function sortedGoods(goods, { sortField }) {
+  const copiedGoods = [...goods];
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  switch (sortField) {
+    case SORT_FIELD_BY_ALPHABET:
+      copiedGoods.sort((good1, good2) => good1.localeCompare(good2));
+      break;
+
+    case SORT_FIELD_BY_LENGTH:
+      copiedGoods.sort((good1, good2) => good1.length - good2.length);
+      break;
+
+    default:
+      break;
+  }
+
+  return copiedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+
+  const [reversedField, setReversedField] = useState(false);
+
+  const reset = () => {
+    setSortField('');
+    setReversedField(false);
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  let visibleGoods = sortedGoods(goodsFromServer, { sortField });
+
+  if (reversedField) {
+    visibleGoods = [...visibleGoods].reverse();
+  }
+
+  return (
+    <div className=" section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {'is-light' : sortField !== SORT_FIELD_BY_ALPHABET})}
+          onClick={() => setSortField(SORT_FIELD_BY_ALPHABET)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn('button', 'is-info', {'is-light' : sortField !== SORT_FIELD_BY_LENGTH})}
+          onClick={() => setSortField(SORT_FIELD_BY_LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button', {'is-light' : !reversedField})}
+          onClick={() => setReversedField(!reversedField)}
+        >
+          Reverse
+        </button>
+
+        {(sortField || reversedField) && (
+          <button
+            onClick={reset}
+            type="button"
+            className="button is-danger is-light"
+          >
+            Reset
+          </button>
+        )}
+      </div>
+      <ul>
+        {visibleGoods.map(good => (
+          <li data-cy="Good"
+          key={good}>
```

Comment:
```suggestion
          <li data-cy="Good" key={good}>
```
correct formatting ⬆️